applications/TopologyOptimizationApplication/custom_utilities/topology_updating_utilities.h

```diff
@@ -0,0 +1,342 @@
+//                   Eric Gonzales
+//
+// ==============================================================================
+
+#if !defined(KRATOS_TOPOLOGY_UPDATING_UTILITIES_H_INCLUDED)
+#define  KRATOS_TOPOLOGY_UPDATING_UTILITIES_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+
+
+// Application includes
+#include "topology_optimization_application.h"
+#include "structure_response_function_utilities.h"
+#include "utilities/builtin_timer.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Solution utility that updates response values for next iteration.
+/** Detail class definition.
+
+ */
+
+class TopologyUpdatingUtilities
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of TopologyUpdatingUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(TopologyUpdatingUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    TopologyUpdatingUtilities( ModelPart& model_part )
+    : mrModelPart(model_part)
+    {
+    }
+
+    /// Destructor.
+    virtual ~TopologyUpdatingUtilities()
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+    // --------------------------------- UPDATE DENSITIES  -----------------------------------------------------------------------------------------
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Finds the value of the X_PHYS (density) and updates it into the optimization problem
+    void UpdateDensitiesUsingOCMethod( char update_type[], double volfrac, double greyscale , double OptItr , double qmax)
+    {
+        KRATOS_TRY;
+
+        if ( strcmp( update_type , "oc_algorithm" ) == 0 ){
+            BuiltinTimer timer;
+            KRATOS_INFO("[TopOpt]") << "  Optimality Criterion Method (OC) chosen to solve the optimization problem" << std::endl;
+
+            // Check if Grey Scale Filter should be used
+            double q = 1;
+            if (greyscale == 1)
+            {
+                if (OptItr < 10)
+                    q = 1;
+                else
+                    q = std::min(qmax, 1.1*q);
+
+                KRATOS_INFO("[TopOpt]") << "  Grey Scale Filter activated, q = " << q << std::endl;
+            }
+            else
+            {
+                KRATOS_INFO("[TopOpt]") << "  Grey Scale Filter deactivated, q = " << q << std::endl;
+            }
+
+            // Update Densities procedure
+            double l1    = 0.0;
+            double l2    = 1.0e12; ///10000000000.0
```

Comment:
Hardcoded?

Reply:
For the moment yes. We would like to implement this in the future.

Reply:
Okay